politeiawww/wsdcrdata.go

```diff
@@ -90,9 +100,73 @@ func (w *wsDcrdata) unsubFromAddr(address string) error {
 	return nil
 }
 
-// newWSDcrdata return a new wsDcrdata context.
-func newWSDcrdata() (*wsDcrdata, error) {
-	// Init websocket client
+// subToNewBlock subscribes to dcrdata events for new blocks
+func (w *wsDcrdata) subToNewBlock() error {
+	if w.isSubscribed(eventNewBlock) {
+		return errDuplicateSub
+	}
+	_, err := w.client.Subscribe(eventNewBlock)
+	if err != nil {
+		return fmt.Errorf("failed to subscribe: %v", err)
+	}
+	w.addSub(eventNewBlock)
+	log.Debugf("wsDcrdata subscribed to new block")
+	return nil
+}
+
+// reconnect stops the current client, initializes a new one, and subscribes
+// to each of the subscriptions that the previous client was subscribed to.
```

Comment:
If you're passing in a channel the comment needs to explain what its being used for and when messages are emitted to it.